docs/enhancements/20190815-zone-aware-routing.md

```diff
@@ -58,43 +58,52 @@ Arguably inter-zone network latency should also be better than cross zone.
+Then Lua balancer when picking an endpoint will try to pick zone-local endpoint first and
+if there is no zone-local endpoint then it will fall back to current behaviour.
 
-This feature at least in the beginning should be optional since it is going to make it harder to reason about the load balancing and not everyone might want that.
+This feature at least, in the beginning, should be optional since it is going to make it harder to reason about the load balancing and not everyone might want that.
 
 **How does controller know what zone it runs in?**
-We can have the pod spec do pass node name using downward API as an environment variable.
-Then on start controller can get node details from the API based on node name. Once the node details is obtained
-we can extract the zone from `failure-domain.beta.kubernetes.io/zone` annotation. Then we can pass that value to Lua land through Nginx configuration
+We can have the pod spec pass the node name using downward API as an environment variable.
+Then on start controller can get node details from the API based on the node name.
+Once the node details are obtained
+we can extract the zone from `failure-domain.beta.kubernetes.io/zone` annotation.
+Then we can pass that value to Lua land through Nginx configuration
 when loading `lua_ingress.lua` module in `init_by_lua` phase.
 
 **How do we extract zones for endpoints?**
 We can have the controller watch create and update events on nodes in the entire cluster and based on that keep the map of nodes to zones in the memory.
 And when we generate endpoints list, we can access node name using `.subsets.addresses[i].nodeName`
 and based on that fetch zone from the map in memory and store it as a field on the endpoint.
-__This solution assumes `failure-domain.beta.kubernetes.io/zone`__ annotation does not change until the end of node's life. Otherwise we have to
+__This solution assumes `failure-domain.beta.kubernetes.io/zone`__ annotation does not change until the end of the node's life. Otherwise, we have to
 watch update events as well on the nodes and that'll add even more overhead.
 
-Alternatively, we can get the list of nodes only when there's no node in the memory for given node name. This is probably a better solution
-because then we would avoid watching for API changes on node resources. We can eagrly fetch all the nodes and build node name to zone mapping on start.
-And from thereon sync it during endpoints building in the main event loop iff there's no entry exist for the node of an endpoint.
+Alternatively, we can get the list of nodes only when there's no node in the memory for the given node name. This is probably a better solution
+because then we would avoid watching for API changes on node resources. We can eagerly fetch all the nodes and build node name to zone mapping on start.
+And from thereon sync it during endpoint building in the main event loop IFF there's no existing entry for the node of an endpoint.
```

Comment:
```suggestion
From thereon,  it will sync during endpoint building in the main event loop if there's no existing entry for the node of an endpoint.
```

Reply:
I'm not a huge fan of `thereon`, is it really preferred over `there on`?

Reply:
there on is probably better

Reply:
```suggestion
From there on,  it will sync during endpoint building in the main event loop if there's no existing entry for the node of an endpoint.
```